core/src/main/java/io/grpc/transport/Http2ClientStream.java

```diff
@@ -136,8 +136,7 @@ protected void transportDataReceived(Buffer frame, boolean endOfStream) {
       if (endOfStream) {
         // This is a protocol violation as we expect to receive trailers.
         transportError = Status.INTERNAL.withDescription("Recevied EOS on DATA frame");
-        frame.close();
-        inboundTransportError(transportError);
+        inboundTransportError(transportError, true);
```

Comment:
That's because request() needed to be called in order to read through the partial message. I'll send out a PR that reverts the stop delivery and fixes the test.